### PR TITLE
Fixed exception trying to access oldVal.pageCount in manual pagination

### PR DESCRIPTION
**Pull request by rjavier-trimbler**
## Description

- Fixed issue when the manualPaginationOptions value is undefined and the property is set to something else. Code was throwing an exception trying to access oldVal.pageCount value in the conditional statement.

- Fixed typo setting manualPagination option instead of manualSorting in the onManualSortOptionsChange watch.

References #1908 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-table/modus-table.tsx [hunk 1]
```diff
@@ -168,19 +172,19 @@ export class ModusTable {
   @Watch('manualSortingOptions') onManualSortOptionsChange(
     newVal: ModusTableManualSortingOptions,
     oldVal: ModusTableManualSortingOptions
-  ) {
+  )  {
     if (newVal?.currentSortingState.length === 0) {
       if (oldVal && oldVal.currentSortingState.length > 0) {
-        this.tableCore.setOptions('manualPagination', true);
```
Comment (rjavier-trimbler) [inline stencil-workspace/src/components/modus-table/modus-table.tsx:174]: we found this issue when testing the manual sorting in our product.

File: stencil-workspace/src/components/modus-table/parts/modus-table-pagination.tsx [hunk 2]
```diff
@@ -13,16 +13,18 @@ export const ModusTablePagination: FunctionalComponent<ModusTablePaginationProps
...
 
   let pageSize = pageSizeState;
+  let pageIndex = pageIndexState;
   let totalCount = data.length ?? 0;
   let activePage = 1;
 
   if (manualPaginationOptions) {
     const { totalRecords, currentPageSize } = manualPaginationOptions;
     activePage = manualPaginationOptions?.currentPageIndex ?? activePage;
+    pageIndex = activePage - 1;
```
Comment (rjavier-trimbler) [inline stencil-workspace/src/components/modus-table/parts/modus-table-pagination.tsx:27]: fixing issue showing always the range 1-{pageSize} with manual pagination

File: stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx [hunk 3]
```diff
@@ -61,8 +61,9 @@ function initializeTable(columns, data, pageSizeList, toolbarOptions, displayOpt
   modusTable.manualPaginationOptions = ${JSON.stringify(manualPaginationOptions)};
   modusTable.manualSortingOptions = ${JSON.stringify(manualSortingOptions)};
 
+  var globalData = ${JSON.stringify(data)};
```
Comment (rjavier-trimbler) [inline stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx:64]: Using 50 rows of input to test manual pagination properly. Updating code below to load only the first page.

File: stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx [hunk 4]
```diff
@@ -672,8 +678,7 @@ export const CheckboxRowSelection = Template.bind({});
 CheckboxRowSelection.args = {
   ...DefaultArgs, rowSelection: true, rowSelectionOptions: {
     multiple: true,
-    subRowSelection: true,
-    preSelectedRows:["0"]
+    subRowSelection: true
```
Comment (cjwinsor) [inline stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx:681]: Is this change intentional?

File: stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx [hunk 5]
```diff
@@ -633,15 +639,15 @@ export const Pagination = Template.bind({});
 Pagination.args = { ...DefaultArgs, pagination: true, data: makeData(50), pageSizeList: [5, 10, 50] };
 
 export const ManualPagination = Template.bind({});
-
 ManualPagination.args = {
   ...DefaultArgs,
   pagination: true,
+  data: makeData(50),
```
Comment (cjwinsor) [inline stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx:645]: Are these changes intentional

**Review by cjwinsor — CHANGES_REQUESTED** [review]

**Comment by cjwinsor** [comment]
@rjavier-trimbler Given that this issue wasn't uncovered due to missing test scenarios. I am asking that more test scenarios be added to cover various cases with manual pagination.  Most specifically, any tests that would have caught this issue.

File: stencil-workspace/src/components/modus-table/modus-table.tsx [hunk 6]
```diff
@@ -168,19 +172,19 @@ export class ModusTable {
   @Watch('manualSortingOptions') onManualSortOptionsChange(
     newVal: ModusTableManualSortingOptions,
     oldVal: ModusTableManualSortingOptions
-  ) {
+  )  {
```
[the remark this answers is not in the document]
Comment (rjavier-trimbler) [reply stencil-workspace/src/components/modus-table/modus-table.tsx]: done!

File: stencil-workspace/src/components/modus-table/modus-table.tsx [hunk 7]
```diff
@@ -154,11 +154,15 @@ export class ModusTable {
...
-      newVal.currentPageSize !== oldVal.currentPageSize
+      newVal?.pageCount !== oldVal?.pageCount ||
+      newVal?.currentPageIndex !== oldVal?.currentPageIndex ||
+      newVal?.currentPageSize !== oldVal?.currentPageSize
     ) {
-      this.tableCore.setOptions('pageCount', newVal.pageCount);
+      this.tableCore?.setOptions('pageCount', newVal.pageCount);
+      this.tableCore?.setState('pagination', {
+        pageIndex: newVal.currentPageIndex - 1,
+        pageSize: newVal.currentPageSize
```
[the remark this answers is not in the document]
Comment (rjavier-trimbler) [reply stencil-workspace/src/components/modus-table/modus-table.tsx]: done!

**Review by cjwinsor — CHANGES_REQUESTED** [review]
Manually testing the property changes expressed in the e2e tests exposes something strange
```html
    <modus-table></modus-table>
    <script>
      function sleep(ms) {
          return new Promise(resolve => setTimeout(resolve, ms));
      }

      async function updateTableProperties() {
          const table = document.querySelector('modus-table');

          table.columns = [
              {
                header: 'Mock Column One',
                accessorKey: 'mockColumnOne',
                id: 'mock-column-one',
                dataType: 'text',
                footer: 'Total',
              },
              {
                header: 'Mock Column Two',
                accessorKey: 'mockColumnTwo',
                id: 'mock-column-two',
                dataType: 'integer',
                showTotal: true,
              },
            ];
            table.data = [
              {
                mockColumnOne: 'Mock Data One 1',
                mockColumnTwo: 330160,
              },
              {
                mockColumnOne: 'Mock Data One 2',
                mockColumnTwo: 900293,
              },
            ];
          table.manualPaginationOptions = {};
          table.pagination = false;

          // Wait for 5 seconds
          await sleep(5000);

          table.pagination = true;
      }
      document.addEventListener('DOMContentLoaded', updateTableProperties);
    </script>
```

image

Same result if I do this after the sleep
```js
          table.manualPaginationOptions = {
              currentPageIndex: undefined,
              currentPageSize: undefined,
              pageCount: undefined,
              totalRecords: undefined,
          };
          table.pageSizeList = [5, 10, 50];
          table.pagination = true;
```

I'll need to consider how this should behave and I still need to test this same scenario with the `main` branch to see if it will cause the original issue the bug was opened for.

**Comment by rjavier-trimbler** [comment]
> Manually testing the property changes expressed in the e2e tests exposes something strange
> 
> ```
>     <modus-table></modus-table>
>     <script>
>       function sleep(ms) {
>           return new Promise(resolve => setTimeout(resolve, ms));
>       }
> 
>       async function updateTableProperties() {
>           const table = document.querySelector('modus-table');
> 
>           table.columns = [
>               {
>                 header: 'Mock Column One',
>                 accessorKey: 'mockColumnOne',
>                 id: 'mock-column-one',
>                 dataType: 'text',
>                 footer: 'Total',
>               },
>               {
>                 header: 'Mock Column Two',
>                 accessorKey: 'mockColumnTwo',
>                 id: 'mock-column-two',
>                 dataType: 'integer',
>                 showTotal: true,
>               },
>             ];
>             table.data = [
>               {
>                 mockColumnOne: 'Mock Data One 1',
>                 mockColumnTwo: 330160,
>               },
>               {
>                 mockColumnOne: 'Mock Data One 2',
>                 mockColumnTwo: 900293,
>               },
>             ];
>           table.manualPaginationOptions = {};
>           table.pagination = false;
> 
>           // Wait for 5 seconds
>           await sleep(5000);
> 
>           table.pagination = true;
>       }
>       document.addEventListener('DOMContentLoaded', updateTableProperties);
>     </script>
> ```
> 
> image
> 
> Same result if I do this after the sleep
> 
> ```js
>           table.manualPaginationOptions = {
>               currentPageIndex: undefined,
>               currentPageSize: undefined,
>               pageCount: undefined,
>               totalRecords: undefined,
>           };
>           table.pageSizeList = [5, 10, 50];
>           table.pagination = true;
> ```
> 
> I'll need to consider how this should behave and I still need to test this same scenario with the `main` branch to see if it will cause the original issue the bug was opened for.

In this case  we're setting an empty object and then changing to the correct properties after the timeout is over, for this particular case we have included to re-render the table again.

**Review by cjwinsor — APPROVED** [review]

**Pull request merged**